connect/runtime/src/test/java/org/apache/kafka/connect/integration/MonitorableSinkConnector.java

```diff
@@ -0,0 +1,169 @@
+                taskAvailable.countDown();
+            }
+        }
+
+        public MonitorableSinkTask task() {
+            try {
+                log.debug("Waiting on task {}", taskName);
+                taskAvailable.await(MAX_WAIT_FOR_TASK_DURATION_MS, TimeUnit.MILLISECONDS);
+                log.debug("Found task!");
+            } catch (InterruptedException e) {
+                throw new ConnectException("Could not find task for " + taskName, e);
+            }
+
+            return taskRef.get();
+        }
+    }
+
+    public static Handle taskInstances(String taskId) {
+        return HANDLES.computeIfAbsent(taskId, connName -> new Handle(taskId));
+    }
+
+    @Override
+    public void start(Map<String, String> props) {
+        connectorName = props.get("name");
+        expectedRecordsStr = props.get(EXPECTED_RECORDS);
+        log.info("Starting connector {}", props.get("name"));
+    }
+
+    @Override
+    public Class<? extends Task> taskClass() {
+        return MonitorableSinkTask.class;
+    }
+
+    @Override
+    public List<Map<String, String>> taskConfigs(int maxTasks) {
+        List<Map<String, String>> configs = new ArrayList<>();
+        for (int i = 0; i < maxTasks; i++) {
+            Map<String, String> config = new HashMap<>();
+            config.put("task.id", connectorName + "-" + i);
+            config.put("required", "dummy-val");
+            config.put(EXPECTED_RECORDS, expectedRecordsStr);
+            configs.add(config);
+        }
+        return configs;
+    }
+
+    // [2018-09-25 11:23:34,562] DEBUG Waiting on task simple-conn-0 (org.apache.kafka.connect.integration.MonitorableSinkConnector:68)
+    // [2018-09-25 11:23:36,058] DEBUG Found task! (org.apache.kafka.connect.integration.MonitorableSinkConnector:70)
+
+    @Override
+    public void stop() {
+
+    }
+
+    @Override
+    public ConfigDef config() {
+        return new ConfigDef();
+    }
+
+    public static class MonitorableSinkTask extends SinkTask {
+
+        private String taskId;
+        private int expectedRecords;
+        private CountDownLatch latch;
+
+        @Override
+        public String version() {
+            return "unknown";
+        }
+
+        @Override
+        public void start(Map<String, String> props) {
+            log.debug("Starting task {}", context);
+            taskId = props.get("task.id");
+            expectedRecords = Integer.parseInt(props.get(EXPECTED_RECORDS));
+            taskInstances(taskId).task(this);
+            latch = new CountDownLatch(expectedRecords);
+        }
+
+        @Override
+        public void put(Collection<SinkRecord> records) {
+            for (SinkRecord rec : records) {
+                latch.countDown();
+                log.debug("Obtained record: {} at {}", rec.value(), context);
+            }
+        }
+
+        @Override
+        public void stop() {
+            log.info("Removing {}", HANDLES.remove(taskId));
```

Comment:
Woah, I almost missed this important call to `remove(taskId)` since it's within a log message. Not sure this is a good idea.

This won't work if the # of tasks is > 1, since the first task to be stopped will remove the Handle for the connector.

Reply:
Wouldn't `taskId` be different for each task?